File: Your-Space-Backend/YourSpace.Data/Entities/Group.cs
using System.ComponentModel.DataAnnotations;

namespace YourSpace.Data.Entities;

public class Group
{
    public int Id { get; set; }

    public required string OwnerUserId { get; set; }

    [MaxLength(200)]
    public required string Name { get; set; }

    [MaxLength(200)]
    public string? NameAr { get; set; }

    public DateTime? DeletedAt { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public AppUser Owner { get; set; } = null!;
}

File: Your-Space-Mobile/lib/core/mock/entities/group.dart
import 'package:equatable/equatable.dart';

class Group extends Equatable {
  final int id;
  final String name;
  final String? nameAr;
  final DateTime createdAt;

  const Group({
    required this.id,
    required this.name,
    this.nameAr,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [id, name, nameAr, createdAt];
}
